site2/docs/functions-package.md

```diff
@@ -0,0 +1,524 @@
+    >
+    > - Function interface provided by Java 8: `java.util.function.Function`
+    >
+    > - Pulsar Function interface: `org.apache.pulsar.functions.api.Function`
+    >
+    > The main difference between the two interfaces is that the `org.apache.pulsar.functions.api.Function` interface provides the context interface. When you write a function and want to interact with it, you can use context to obtain a wide variety of information and functionality for Pulsar Functions.
+    > 
+    > **Example** 
+    >
+    > This example uses `org.apache.pulsar.functions.api.Function` interface with context.
+    >
+    > ```
+    > package org.example.functions;
+    >
+    > import org.apache.pulsar.functions.api.Context;
+    > import org.apache.pulsar.functions.api.Function;
+    > 
+    > import java.util.Arrays;
+    >
+    > public class WordCountFunction implements Function<String, Void> {
+    >    // This function is invoked every time a message is published to the input topic
+    >    @Override
+    >    public Void process(String input, Context context) throws Exception {
+    >       Arrays.asList(input.split(" ")).forEach(word -> {
+    >           String counterKey = word.toLowerCase();
+    >           context.incrCounter(counterKey, 1);
+    >        });
+    >       return null;
+    >   }
+    > }
+    > ```
+
+3. Package the Java function.
+
+    ```bash
+    mvn package
+    ```
+
+    After the Java function is packaged, a `target` directory is automatically created. Open the `target` directory to see if there is a jar package similar to `java-function-1.0-SNAPSHOT.jar`.
+
+
+4.  Run the Java function.
+
+     (1) Copy the packaged jar file to the Pulsar image.
+
+    ```bash
+    docker exec -it [CONTAINER ID] /bin/bash
+    docker cp <path of java-function-1.0-SNAPSHOT.jar>  CONTAINER ID:/pulsar
+    ```
+
+    (2) Run the Java function using the following command.
+
+    ```bash
+    ./bin/pulsar-admin functions localrun \
+    --classname org.example.test.ExclamationFunction \
+    --jar java-function-1.0-SNAPSHOT.jar \
+    --inputs persistent://public/default/my-topic-1 \
+    --output persistent://public/default/test-1 \
+    --tenant public \
+    --namespace default \
+    --name JavaFunction
+    ```
+
+    The following log indicates that the Java function starts successfully.
+
+    ```text
+    ...
+    07:55:03.724 [main] INFO  org.apache.pulsar.functions.runtime.ProcessRuntime - Started process successfully
+    ...
+    ```
+
+    > **Tip**
+    >
+    >  - For the description about the parameters (for example, `--classname`, `--jar`, `--inputs`, and so on), run the command `./bin/pulsar-admin functions` or see [here](reference-pulsar-admin.md#functions).
+    > 
+    > - If you want to start a function in cluster mode, replace `localrun` with `create` in the command above. The following log indicates that the Java function starts successfully.
+    >
+    >   ```text
+    >   "Created successfully"
+    >   ```
```

Comment:
I see the same tip occurs several times in this doc, if it's important information, you can add it in the body. If you prefer tips, I suggest that we can keep one.
For guides on "Tip/Note", refer to https://developers.google.com/style/notices